core/src/test/scala/shapeless/serialization.scala

```diff
@@ -937,11 +943,11 @@ class SerializationTests {
   def testLazy {
     assertSerializable(Lazy(23))
 
-    assertSerializable(implicitly[Lazy[Generic[Wibble]]])
-    assertSerializable(implicitly[Lazy[Generic1[Box, TC1]]])
+    assertSerializableBeforeAfter(implicitly[Lazy[Generic[Wibble]]])(_.value)
+    assertSerializableBeforeAfter(implicitly[Lazy[Generic1[Box, TC1]]])(_.value)
 
```

Comment:
I'm still puzzled ... this is testing serializability before and after forcing lazy vals? But surely the first  serializing is itself forcing any lazy vals? I guess I'm suspicious because I don't fully understand how there could be a difference here.